### PR TITLE
Fix type for imaplib.IMAP4.error and related

### LINKED ISSUE #2268: imaplib.IMAP4.error is defined as Exception instance in stub
In `imaplib.pyi`, `IMAP4.error` is defined as Exception instance:
```
class IMAP4:
    error: Exception
    ...
```

but actually, it is a Exception **class**:
```
class IMAP4:
    ...
    class error(Exception): pass    # Logical errors - debug required
    ...
```

This mismatch cause type error (mypy say `error: Exception type must be derived from BaseException`) on code below:
```
import imaplib

try:
    ...
except imaplib.IMAP4.error:
    pass
```

### PR DESCRIPTION
**Pull request by nakamuray**
fix #2268

### CODE REVIEW

File: stdlib/2and3/imaplib.pyi [hunk 1]
```diff
@@ -6,15 +6,15 @@ import sys
...
+from typing import Any, Callable, Dict, IO, List, Optional, Pattern, Text, Tuple, Type, Union
 
 CommandResults = Tuple[str, List[Any]]
 
 
 class IMAP4:
-    error: Exception
-    abort: Exception
-    readonly: Exception
+    error: Type[Exception] = ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/imaplib.pyi]: Does it work you follow the implementation more closely and write `class error(Exception): ...` here?

Comment (nakamuray) [reply stdlib/2and3/imaplib.pyi]: I tried, but that broke pytype_test.py. Error below:

```
stdlib/2and3/imaplib.pyi: ParseError: syntax error, unexpected CLASS, expecting DEF or '@'
```

Comment (JelleZijlstra) [reply stdlib/2and3/imaplib.pyi]: Oh well. In that case, the current code should be OK.

**Comment by JelleZijlstra** [comment]
Thanks for your PR!

**Pull request merged**